x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/type/EsqlDataTypeConverter.java

```diff
@@ -369,6 +370,9 @@ public static DataType commonType(DataType left, DataType right) {
             if (left == TEXT || right == TEXT) {
                 return TEXT;
             }
+            if (left == SEMANTIC_TEXT || right == SEMANTIC_TEXT) {
+                return TEXT;
```

Comment:
`SEMANTIC_TEXT` does not perform any analysis on the raw text, would `KEYWORD` be more appropriate in this case?